sound/soc/sof/intel/hda-dsp.c

```diff
@@ -212,6 +212,10 @@ int hda_dsp_enable_core(struct snd_sof_dev *sdev, unsigned int core_mask)
 {
 	int ret;
 
+	/* return if core is already enabled */
+	if (hda_dsp_core_is_enabled(sdev, core_mask))
+		return 0;
+
```

Comment:
does this have anything to do with this patchset? This isn't really related to multi-core, is it?

Reply:
@plbossart it is. When we have load multiple pipelines scheduled to run on the same core, we will be trying to power it up over and over again. I added this to prevent that.